CIPs/cip-0044.md

```diff
@@ -0,0 +1,76 @@
+author: Mikerah Quintyne-Collins (@Mikerah), Kob Gurkan (@kobigurkan), Philipp Jovanovic (@daeinar)
+discussions-to: https://github.com/celo-org/celo-proposals/issues/270
+status: Draft
+type: Standards
+category (only required for Standards Track): <Ring 0, 1, 2, 3>
+created: 2021-08-10
+license: Apache 2.0
+---
+
+## Simple Summary
+The goal of this CIP is to specify *left/right constrained pseudorandom functions (PRFs)*, a cryptographic primitive introduced in [BW13](BW13), for use in Celo protocols. 
+
+## Abstract
+Celo is a mobile-first blockchain protocol that seeks to facilitate making cryptocurrency payments from mobile devices. As such, Celo has developed a series of subprotocols for enabling users to utilize their phone numbers for various aspects of the Celo protocol. Since phone numbers are tied directly to users' identities, it is of paramount importance to put appropriate protection mechanisms into place that ensure users' privacy. To this end, cLabs has developed and deployed the [Phone Number Privacy](PNP) subprotocol on Celo and is currently working on several additional enhancements. Some of these mechanisms rely on so-called *left/right constrained PRFs*, a cryptographic primitive that is useful to build privacy-preserving identity-based protocols.
+
+This CIP specifies the parameters and algorithms for left/right constrained PRFs for use in Celo protocols.
+
+## Motivation
+This CIP introduces left/right constrained PRFs that can be used to develop improved privacy-protection mechanisms for users on the Celo blockchain.
+
+## Specification
+We follow the specification of (left/right) constrained PRFs as introduced in [BW13](BW13).
+
+### Constrained PRFs
+Given a key space `K`, a domain `X`, range `Y`, let `F` be a pseudorandom function that takes as inputs `(k, x)` for `k` in `K` and `x` in `X` and returns `F(k, x)` in `Y`. 
+
+We say that `F` is constrained with respect to a set of constraints `S` if there is a new key space `K_c`, such that we can define the following functions:
+- `F.constrain(k, S)` which provides a constrained key `k_S` in `K_c`. This key `k_S` enables the evaluation of `F(k, x)` for all `x` in `S` and no other `x`.
+- `F.eval(k_S, x)` which outputs `F(k, x)` if `x` is in `S` and `nil` otherwise.
+
+In practice, `S` is usually a set of predicates `PP` such that for any `p` in `PP`, we have `F.eval(k_p, x) = F(k, x)` if `p(x) = 1` and `F.eval(k_p, x) = nil` otherwise.
+
+### Left/right constrained PRFs
+Let `F: K x X^2 -> Y` be a PRF. Then, `F` is a left/right constrained PRF if `F` is constrained by the following predicates `P_LR = {(p_wL, p_wR), w in X}` where `p_wL(x,y) = 1` iff `x = w` and `p_wR(x,y) = 1` iff `y = w`. Left/right constrained PRFs have the property such that for all points in `(w, y)` in `X^2`, `F(k, (w, y))` can be evaluated at all the points in which the left side is fixed to `w`. Similarly for right constrained PRFs. 
```

Comment:
* s/following predicates/predicate family
* `P_LR = {p_wL, p_wR: w in X}`-- drop the parens because they're separate predicates.
* s/`F(k, (w, y))`/`F(k_p_wL, (w, y))`

Reply:
Please continue the discussion here: https://github.com/celo-org/celo-proposals/issues/270